src/test/tv/codely/scala_intro_examples/lesson_09_oop/exercise_solutions/joan_miralles_solution/EmailTest.scala

```diff
@@ -0,0 +1,46 @@
+package tv.codely.scala_intro_examples.lesson_09_oop.exercise_solutions.joan_miralles_solution
+
+import java.security.InvalidParameterException
+
+import org.scalatest.{Matchers, WordSpec}
+
+class EmailTest extends WordSpec with Matchers {
+
+  private val validEmail = "some@email.com"
+  private val anotherValidEmail = "a@e.t"
+  private val invalidEmail = "some.email.com"
+  private val anotherInvalidEmail = "some@email"
+
+  "Email Class" should {
+    "provide an apply method in the companion object in order to construct new Email from complete string" in {
+      "Email(completeEmail = validEmail)" should compile
+    }
+
+    "return a new Email instance given a complete email" in {
+      val email = Email(validEmail)
+      assert(email.isInstanceOf[Email])
+      email.local shouldBe "some"
+      email.domain shouldBe "email.com"
+    }
+
+    "return a new Email instance given a complete email (2)" in {
```

Comment:
Naming: Instead of just specifying a number `(2)`, we could explain why are we adding a new test case. Which are the particularities of that case that make you add an specific test case for it?

Suggestion: `return a new Email instance given a email address string which doesn't contain a valid TLD`? (reading it that way, I don't know if our program should take into account these kind of scenarios as a valid email addresses :S)